### PR TITLE
join client CA bundles

### PR DESCRIPTION
**Pull request by deads2k**
Last commit grabs client CA bundles from disparate parts of the auth config and makes a pool that contains all of them.

I suspect a rebase broke this because of ordering.  I'll keep these separate to make it easier for me to debug.

### CODE REVIEW

File: pkg/genericapiserver/config.go [hunk 1]
```diff
@@ -301,17 +303,37 @@ func (c *Config) ApplyInsecureServingOptions(insecureServing *options.ServingOpt
...
 	if o == nil || o.PasswordFile == nil {
-		return c
+		return c, nil
 	}
 
-	if o.ClientCert != nil && c.SecureServingInfo != nil {
-		c.SecureServingInfo.ClientCA = o.ClientCert.ClientCA
+	if c.SecureServingInfo != nil {
+		if c.SecureServingInfo.ClientCA == nil {
+			c.SecureServingInfo.ClientCA = x509.NewCertPool()
```
Comment (liggitt) [inline pkg/genericapiserver/config.go]: only initialize this if we're going to populate it... a non-nil cert pool makes the server prompt for client certs, and sending an empty CA list makes clients try to send any client cert they have

**Comment by liggitt** [comment]
nit on initialization of client certpool, last commit LGTM otherwise

File: pkg/genericapiserver/config.go (hunk 1 above)
Comment (deads2k) [inline pkg/genericapiserver/config.go]: > only initialize this if we're going to populate it... a non-nil cert pool makes the server prompt for client certs, and sending an empty CA list makes clients try to send any client cert they have

yikes

**Comment by deads2k** [comment]
@k8s-bot unit test this: issue #37961

**Comment by deads2k** [comment]
@k8s-bot unit test this

**Pull request merged**